src/util/text-tracks.js

```diff
@@ -105,165 +103,121 @@ const deprecateOldCue = function(cue) {
   });
 };
 
-export const durationOfVideo = function(duration) {
-  let dur;
-
-  if (isNaN(duration) || Math.abs(duration) === Infinity) {
-    dur = Number.MAX_VALUE;
-  } else {
-    dur = duration;
-  }
-  return dur;
-};
 /**
- * Add text track data to a source handler given the captions and
- * metadata from the buffer.
+ * Add metadata text track data to a source handler given an array of metadata
  *
- * @param {Object} sourceHandler the virtual source buffer
- * @param {Array} captionArray an array of caption data
- * @param {Array} metadataArray an array of meta data
+ * @param {Object}
+ *   @param {Object} inbandTextTracks the inband text tracks
+ *   @param {Array} metadataArray an array of meta data
+ *   @param {Number} timestampOffset the timestamp offset of the source buffer
+ *   @param {Number} videoDuration the duration of the video
  * @private
  */
-export const addTextTrackData = function(sourceHandler, captionArray, metadataArray) {
-  let Cue = window.WebKitDataCue || window.VTTCue;
-
-  if (captionArray) {
```

Comment:
This is/was `addCaptionData`, not sure why we repeated the logic here, but it was removed and is now done in `segment-loader`.